src/control/LatexController.cpp

```diff
@@ -333,6 +448,12 @@ TexImage* LatexController::loadRendered()
 	Path pdfPath = texTmp + "/tex.pdf";
 	GError* err = NULL;
 
+	if (!pdfPath.exists())
+	{
+		g_warning("LaTeX preview PDF file does not exist");
```

Comment:
Also show this to the user.

Reply:
Actually, a huge portion of this file could be refactored and simplified, which would make this check unnecessary.

Reply:
You may either clean this up in this commit or in a follow-up. Thats for you to decide. I'm happy either way.

Reply:
Refactored in one of the commits I pushed today.